core/src/DynamicSchema.ts

```diff
@@ -149,17 +149,19 @@ async function createDynamicSchema(
 
   const { schemaName, schemaAlias } = props;
   const newSchema = new MetaSchema(context, schemaName, schemaAlias, version.readVersion, version.writeVersion, version.minorVersion);
-
-  const loader = new IModelSchemaLoader(db);
-  const bisSchema = loader.getSchema("BisCore");
+  const bisSchema = await tryGetSchema(db, "BisCore");
   await context.addSchema(newSchema);
-  await context.addSchema(bisSchema);
-  await (newSchema as MutableSchema).addReference(bisSchema); // TODO remove this hack later
+  if (bisSchema != undefined) {
+    await context.addSchema(bisSchema);
+    await (newSchema as MutableSchema).addReference(bisSchema); // TODO remove this hack later
```

Comment:
Do you know what this "hack" is about? It's later, so can we consider something better while we're in the area?

Reply:
@jffmarker This comment was there since the beginning.

D:\source_trees\discard-by-2025\pcf-OLD\pcf\core\src>git annotate DynamicSchema.ts | findstr hack
e0c55992        (Zachary Zhu    2021-06-14 17:34:59 -0400       157)  await (newSchema as MutableSchema).addReference(bisSchema); // TODO remove this hack later

D:\source_trees\discard-by-2025\pcf-OLD\pcf\core\src>git log e0c55992  
commit e0c559921bc66a366a47d8731bec72cd44fe1e2c
Author: Zachary Zhu <Zachary.Zhu@bentley.com>
Date:   Mon Jun 14 17:34:59 2021 -0400

    initial commit

Reply:
I tried to remove "as MutableSchema" to see if that was the hack and sure enough the "addReference" method is protected on the MetaSchema and therefore the type assertion is done.  This is very likely the hack as you can see in the documentation that popped up for "MutableSchema" even uses the expression "hackish"!
![MutableSchema](https://github.com/iTwin/pcf/assets/66284944/5d26d6ed-7700-4430-82f7-ba1c242612d7)

Reply:
@jffmarker  I created a new backlog item for this. https://dev.azure.com/bentleycs/iModelTechnologies/_workitems/edit/1409051